drivers/i2s/i2s_mcux_sai.c

```diff
@@ -314,17 +316,19 @@ static void i2s_dma_tx_callback(const struct device *dma_dev,
 			strm->state = I2S_STATE_ERROR;
 			goto disabled_exit_no_drop;
 		}
-		dma_start(dev_data->dev_dma, strm->dma_channel);
 
 		if (blocks_queued ||
 			       (strm->free_tx_dma_blocks < MAX_TX_DMA_BLOCKS)) {
+			dma_start(dev_data->dev_dma, strm->dma_channel);
 			goto enabled_exit;
 		} else {
 			/* all DMA blocks are free but no blocks were queued */
-			if (strm->state == I2S_STATE_STOPPING) {
-				/* TX queue has drained */
-				strm->state = I2S_STATE_READY;
-				LOG_DBG("TX stream has stopped");
+			if (strm->state == I2S_STATE_STOPPING || blocks_queued == 0) {
+				/* Received a STOP/DRAIN trigger */
+				k_work_reschedule(&strm->difer_work,
+				  K_MSEC(1));
```

Comment:
Hi @hakehuang can you help to answer the question about i2s_mcux_sai.c changes.

Reply:
this is to ensure the signal is out for iws master. it does not hurt, because at the end the sai out clock will be closed. you may create a config for this

Reply:
I still have concerns with this arbitrary 1ms delay.  I will take this offline with you and Qiang.